memcached/memcached.go

```diff
@@ -122,15 +122,21 @@ func (c *CacheClient) AddItem(key string, value interface{}, expiration int32) (
 func (c *CacheClient) UpdateItem(key string, value interface{}, expiration int32, addIfNotExists bool) (bool, error) {
 	item, err := CreateMemCacheObject(key, value, expiration)
 	if err != nil {
-		return false, err
+		c.logger.LogError("Error occurred while updating item in cache.", err)
+		return false, nil
 	}
 	err = c.client.Replace(item)
 	if err != nil {
 		//unable to find key in cache
 		if addIfNotExists {
-			return c.AddItem(key, value, expiration)
+			val, err := c.AddItem(key, value, expiration)
+			if err != nil {
+				c.logger.LogError("Error occurred while updating item in cache.", err)
+			}
+			return val, nil
 		}
-		return false, err
+		c.logger.LogError("Error occurred while updating item in cache.", err)
```

Comment:
This will cause unnecessary errors when someone tries to update an item which is not in memcache. If addIfNotExists is false, then it is expected that the item won't be added.

Reply:
no, this will be executed, when update fails and adding item also fails